homeassistant/helpers/config_validation.py

```diff
@@ -65,9 +68,36 @@ def boolean(value: Any) -> bool:
     return bool(value)
 
 
-def isfile(value):
+def isfile(value: Any) -> str:
     """Validate that the value is an existing file."""
-    return vol.IsFile('not a file')(value)
+    if value is None:
+        raise vol.Invalid('None is not file')
+    return vol.IsFile('not a file')(str(value))
```

Comment:
`vol.IsFile` uses `os.path.isfile`, but might be good to check read access as well using `os.acces (... R_OK)` like you do below